### DISCUSSION TITLE
How to only create a PR for matchedDependencies with Renovate

### DISCUSSION THREAD

**Discussion by epiglottis** [renovatebot/renovate#9640] [Request Help]
I want to only create a Pull Request for the below specified dependencies. All other dependencies should not be considered. The config below does not do that because I get separate PR's for all dependencies that do not match the rules I specified. For example I get a PR for SimpleInjector. How can I get one PR only for the Dependencies that match, but no other PR's at all without creating explicit exclusion rules for every dependency?

```
  "packageRules": [
    {
      "matchPackagePatterns": [
        "System.*",
        "Microsoft.*",
      ],
      "excludePackageNames": ["Microsoft.CodeAnalysis.FxCopAnalyzers"],
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "groupName": "non major",
      "groupSlug": "non-major"
    }
  ]
```

**Comment by rarkins** [comment]
Your package rule has mistakes, so it's not possible to guess exactly what you want from it. e.g. there is no overlap between the `matchPackagePatterns` value and the `excludePackageNames` value.

Please write out in a simple list exactly what you want to do, and then we can help you with a config.

**Reply by epiglottis** [reply]
Thanks for your quick response. I edited the config.

With this configuration I would expect to get one Pull Request containing all the System.* and Microsoft.* dependencies except for Microsoft.CodeAnalysis.FxCopAnalyzers.
What I actually get is one PR containing all the System.* and Microsoft.* dependencies except for Microsoft.CodeAnalysis.FxCopAnalyzers (which is what I want) but then I also get many pull requests for other dependencies like SimpleInjector or Azure.* which I do not want to regard at all.
My intention is to just regard dependencies specified in matchPackagePatterns and to ignore anything else including Microsoft.CodeAnalysis.FxCopAnalyzers.

**Reply by rarkins** [reply]
The default behavior is that everything is enabled, so you need a rule to *disable* that if that's what you want. 

Rules are matched and executed in order, so I recommend making them verbose for easier understanding.

Here is an example:

```
  "packageRules": [
    {
      "matchPackagePatterns": ["*"],
      "enabled": false
    },
    {
      "matchPackagePatterns": [
        "System.*",
        "Microsoft.*",
      ],
      "excludePackageNames": ["Microsoft.CodeAnalysis.FxCopAnalyzers"],
      "enabled": true
    },
    {
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "groupName": "non major"
    },
    {
      "matchUpdateTypes": ["major"],
      "enabled": false
    }
  ]
```

**Reply by epiglottis** [reply]
Thanks, that solved my problem
